### DISCUSSION TITLE
'PWM' object has no attribute 'init'

### DISCUSSION THREAD

**Discussion by TRadigk** [micropython/micropython#10570] [RP2040 / Pico]
Hi dear developers of micropython,

using my RP2040 with micropython 1.19.1 I found the error message in the title. I don't know if this is a documentation error, or it was intended to be part of the firmware.

code:
```
>>> from machine import Pin, PWM
>>> pin1 = Pin(1)
>>> pwm1 = PWM(pin1)
>>> pwm1.init()
```
result:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'PWM' object has no attribute 'init'
```

Searching for this error I found no corresponding issue. Since the PWM pin will work without calling init(), I don't think it is a critical issue. You will have the insight if it is just an issue of documentation. I referred to: https://docs.micropython.org/en/v1.19.1/library/machine.PWM.html

> PWM.init(*, freq, duty_u16, duty_ns)
> Modify settings for the PWM object. See the above constructor for details about the parameters.

firmware file name: rp2-pico-20220618-v1.19.1.uf2
Board: Raspberry pico (RP2040)

> MicroPython v1.19.1 on 2022-06-18; Raspberry Pi Pico with RP2040
> Type "help()" for more information.

**Comment by robert-hh** [comment]
This is no bug. As documented, not every port supports every module's methods. And the init() method is almost everywhere redundant, because the settings given in the init() call can also be given in the constructor call. Sometimes it is even confusing using the init() method, because in the internal semantics differs between ports & devices. Sometime properties not mentioned in the init() call are not changes, sometimes they are set to default values. This inconsistency grown over years is more of a  problem. Given that, having not init() method can be seen as a benefit.
